Doc/c-api/object.rst

```diff
@@ -6,6 +6,51 @@ Object Protocol
 ===============
 
 
+.. c:function:: PyObject* Py_GetConstant(unsigned int constant_id)
+
+   Get a :term:`strong reference` to a constant.
+
+   Return ``NULL`` if *constant_id* is invalid.
+
+   *constant_id* must be one of these constant identifiers:
+
+   ========================================  ==============
+   Constant Identifier                       Constant Value
+   ========================================  ==============
+   .. c:macro:: Py_CONSTANT_NONE             ``0``
+   .. c:macro:: Py_CONSTANT_FALSE            ``1``
+   .. c:macro:: Py_CONSTANT_TRUE             ``2``
+   .. c:macro:: Py_CONSTANT_ELLIPSIS         ``3``
+   .. c:macro:: Py_CONSTANT_NOT_IMPLEMENTED  ``4``
+   .. c:macro:: Py_CONSTANT_ZERO             ``5``
+   .. c:macro:: Py_CONSTANT_ONE              ``6``
+   .. c:macro:: Py_CONSTANT_EMPTY_STR        ``7``
+   .. c:macro:: Py_CONSTANT_EMPTY_BYTES      ``8``
+   .. c:macro:: Py_CONSTANT_EMPTY_TUPLE      ``9``
+   ========================================  ==============
+
+   Constant values are only given for projects which cannot use the constant
+   identifiers.
```

Comment:
- Use `c:namespace:: NULL` to avoid Sphinx treating the values as `Py_GetConstant.Py_CONSTANT_TRUE` in references, URLs, etc.
- Consider listing the returned Python object -- mainly for links to `Ellipsis` and `NotImplemented` which are not too common.
```suggestion
   .. c:namespace:: NULL

   ========================================  ===== =========================
   Constant Identifier                       Value Returned object
   ========================================  ===== =========================
   .. c:macro:: Py_CONSTANT_NONE             ``0`` :py:data:`None`
   .. c:macro:: Py_CONSTANT_FALSE            ``1`` :py:data:`False`
   .. c:macro:: Py_CONSTANT_TRUE             ``2`` :py:data:`True`
   .. c:macro:: Py_CONSTANT_ELLIPSIS         ``3`` :py:data:`Ellipsis`
   .. c:macro:: Py_CONSTANT_NOT_IMPLEMENTED  ``4`` :py:data:`NotImplemented`
   .. c:macro:: Py_CONSTANT_ZERO             ``5`` ``0``
   .. c:macro:: Py_CONSTANT_ONE              ``6`` ``1``
   .. c:macro:: Py_CONSTANT_EMPTY_STR        ``7`` ``''``
   .. c:macro:: Py_CONSTANT_EMPTY_BYTES      ``8`` ``b''``
   .. c:macro:: Py_CONSTANT_EMPTY_TUPLE      ``9`` ``()``
   ========================================  ===== =========================

   Numeric values are only given for projects which cannot use the constant
   identifiers.

```